src/components/auth/login-form/LoginForm.tsx

```diff
@@ -0,0 +1,116 @@
+import './LoginForm.scss';
+
+import { LoginRequest } from '../../../types/auth/login-request.type';
+import { PRODUCTS_URL_PREFIX } from '../../../constants/api.constants';
+import React from 'react';
+import { setAxiosAccessToken } from '../../../utils/request.utils';
+import { useAuthContext } from '../../../context/AuthContext';
+import { useForm } from 'react-hook-form';
+import { useLogin } from '../../../hooks/auth/useLogin';
+import { useNavigate } from 'react-router-dom';
+import { z } from 'zod';
+import { zodResolver } from '@hookform/resolvers/zod';
+
+const loginSchema = z.object({
+  emailAddress: z
+    .string()
+    .min(3, 'Email is required')
+    .email('Entered email is invalid.'),
+  password: z.string().min(3, 'Password is required'),
+});
+
+const LoginForm: React.FC = () => {
+  const navigate = useNavigate();
+  const { register, handleSubmit, formState } = useForm<LoginRequest>({
+    resolver: zodResolver(loginSchema),
+    mode: 'onTouched',
+  });
+
+  const { login: authLogin, accessToken } = useAuthContext();
+  const { login, isLoading } = useLogin();
+
+  const handleLogin = async (credentials: LoginRequest) => {
+    const response = await login(credentials);
+    if (response) {
+      authLogin(
```

Comment:
**suggestion:** do all the logic to set token and re-route in this `authLogin` method to avoid state issues